PlaywrigthUITests/PageObjects/DemoQAWebTablesPage.cs

```diff
@@ -89,16 +95,126 @@ public async Task VerifyTableRowContent(string headerName = "First Name", string
-        await Assertions.Expect(popup).ToBeVisibleAsync();
+        var isPopupOpened = await RegistrationPopup.IsVisibleAsync();
+        Assert.That(isPopupOpened, "The 'Add' popup is not visible.");
     }
 
     public async Task VerifyFirstNameVisible()
     {
-        var popup = Page.Locator(".modal-content");
-        var firstName = popup.GetByPlaceholder("First Name");
+        var firstName = Page.GetByPlaceholder("First Name");
         await Assertions.Expect(firstName).ToBeVisibleAsync();
     }
     
+    public async Task OpenAddPopup()
+    {
+        await AddButton.ClickAsync();
+        Assert.That(await RegistrationPopup.IsVisibleAsync(), "The 'Add' popup is not visible.");
+    }
+    
+    public async Task FillFirstName (string firstName)
+    {
+        if(firstName == null) throw new Exception("First Name field should not be null.");
+        await FirstNameField.FillAsync(firstName);
+        Assert.That(await FirstNameField.GetAttributeAsync("value") == firstName, $"The First Name field was not filled with: '{firstName}'.");
+    }
+    
+    public async Task FillLastName (string lastName)
+    {
+        if(lastName == null) throw new Exception("Last Name field should not be null.");
+        await LastNameField.FillAsync(lastName);
+        Assert.That(await LastNameField.GetAttributeAsync("value") == lastName, $"The Last Name field was not filled with: '{lastName}'.");
+    } 
+    
+    public async Task FillEmail (string email)
+    {
+        if(email == null) throw new Exception("Email field should not be null.");
+        await EmailField.FillAsync(email);
+        Assert.That(await EmailField.GetAttributeAsync("value") == email, $"The Email field was not filled with: '{email}'.");
+    }
+    
+    public async Task FillAge (string age)
+    {
+        if(age == null) throw new Exception("Age field should not be null.");
+        await AgeField.FillAsync(age);
+        Assert.That(await AgeField.GetAttributeAsync("value") == age, $"The Age field was not filled with: '{age}'.");
+    }
+    
+    public async Task FillSalary (string salary)
+    {
+        if(salary == null) throw new Exception("Salary field should not be null.");
+        await SalaryField.FillAsync(salary);
+        Assert.That(await SalaryField.GetAttributeAsync("value") == salary, $"The Salary field was not filled with: '{salary}'.");
+    }
+    
+    public async Task FillDepartment (string department)
+    {
+        if(department == null) throw new Exception("Department field should not be null.");
+        await DepartmentField.FillAsync(department);
+        Assert.That(await DepartmentField.GetAttributeAsync("value") == department, $"The Department field was not filled with: '{department}'.");
+    }
+
+    public async Task ClickSubmit()
+    {
+        var submit = Page.GetByRole(AriaRole.Button, new() { Name = "Submit" });
+        await submit.ClickAsync();
+    }
+    
+    public async Task<bool> FindRowByEmail(string email)
```

Comment:
it is weird that FindRowByEmail returns bool, but not string or List<string>
This function is not used anymore and should be removed

Reply:
function removed